### PR TITLE
Feature/material

### PR DESCRIPTION
**Pull request by erikeelde**
Removed debug theme. 
Introduces new material design libraries and make some visual changes to the main application.
Transitions main wrench app to koin @joezorry @MarijanGazica 
Removes use of asynctasks for db io operations (moves to coroutines with global scope for now)
Scope functionality is temporarily moved into overflow of actionbar.

screenshot_1546822306

screenshot_1546822311

screenshot_1546822316

### CODE REVIEW

File: wrench-app/src/main/java/com/izettle/wrench/dialogs/booleanvalue/BooleanValueFragment.kt [hunk 1]
```diff
@@ -46,17 +38,17 @@ class BooleanValueFragment : DialogFragment(), Injectable {
         })
 
         return AlertDialog.Builder(requireActivity())
-                .setTitle(R.string.select_scope)
+                .setTitle(".")
```
Comment (mattec92) [inline wrench-app/src/main/java/com/izettle/wrench/dialogs/booleanvalue/BooleanValueFragment.kt:41]: .

File: wrench-app/src/main/java/com/izettle/wrench/dialogs/scope/ScopeFragmentViewModel.kt [hunk 2]
```diff
@@ -27,23 +28,27 @@ constructor(private val scopeDao: WrenchScopeDao) : ViewModel() {
...
     internal fun selectScope(wrenchScope: WrenchScope) {
-        wrenchScope.timeStamp = Date()
-        scopeDao.update(wrenchScope)
+        GlobalScope.launch {
+            wrenchScope.timeStamp = Date()
+            scopeDao.update(wrenchScope)
+        }
     }
 
     @WorkerThread
```
Comment (mattec92) [inline wrench-app/src/main/java/com/izettle/wrench/dialogs/scope/ScopeFragmentViewModel.kt]: These annotations are not accurate any more

Comment (erikeelde) [reply wrench-app/src/main/java/com/izettle/wrench/dialogs/scope/ScopeFragmentViewModel.kt]: 👍 removed! good catch!

File: wrench-app/src/main/java/com/izettle/wrench/dialogs/booleanvalue/BooleanValueFragment.kt (hunk 1 above)
Comment (erikeelde) [reply wrench-app/src/main/java/com/izettle/wrench/dialogs/booleanvalue/BooleanValueFragment.kt:41]: Turns out you need to define a title upon creation to be able to redefine the text in the title at a later time. In this case using R.string.select_scope felt inappropriate. Is "." less bad than before?
Kinda wanna redo those dialogs to look a little nicer. That will probably involve creating a textview where we can set the name of the configuration. 
Especially the one for "select scope" looks off now.

Comment (mattec92) [reply wrench-app/src/main/java/com/izettle/wrench/dialogs/booleanvalue/BooleanValueFragment.kt:41]: 👍 Weird but at least I get why it's a dot now.

File: wrench-app/src/test/java/com/izettle/wrench/provider/WrenchProviderTest.kt [hunk 3]
```diff
@@ -1,43 +1,71 @@
...
+}
 
 @RunWith(AndroidJUnit4::class)
-class WrenchProviderTest : ProviderTestCase2<WrenchProvider>(WrenchProvider::class.java, BuildConfig.CONFIG_AUTHORITY) {
+class WrenchProviderTest : KoinTest {
+
+    private lateinit var wrenchProvider: WrenchProvider
 
-    private var contentResolver: MockContentResolver? = null
+    private var databass: WrenchDatabase? = null
```
Comment (mattec92) [inline wrench-app/src/test/java/com/izettle/wrench/provider/WrenchProviderTest.kt]: Googled databass. Was a bit disappointed. 
http://engelsanchez.net/databass/databass.gif

Comment (erikeelde) [reply wrench-app/src/test/java/com/izettle/wrench/provider/WrenchProviderTest.kt]: Haha. was not expecting anyone to do that. The property has been removed however - it was only added for debugging purposes.

**Pull request merged**